File: examples/ru/07-websocket/node/test_websocket.js
const WebSocket = require('ws');

const ws = new WebSocket('ws://127.0.0.1:8899');

ws.on('open', function open() {
    console.log('✅ Подключено к серверу');
    
    // Тест 1: Простой вывод
    const test1 = {
        code: "print('Hello, World!')"
    };
    console.log('\n📤 Отправка теста 1:', JSON.stringify(test1));
    ws.send(JSON.stringify(test1));
});

let testCount = 0;

ws.on('message', function message(data) {
    const response = JSON.parse(data);
    console.log('\n📥 Получен ответ:');
    console.log('  Success:', response.success);
    console.log('  Output:', response.output);
    if (response.error) {
        console.log('  Error:', response.error);
    }
    
    testCount++;
    
    if (testCount === 1) {
        // Тест 2: Переменные
        const test2 = {
            code: "global x = 10\nglobal y = 20\nprint('Sum:', x + y)"
        };
        console.log('\n📤 Отправка теста 2:', JSON.stringify(test2));
        ws.send(JSON.stringify(test2));
    } else if (testCount === 2) {
        // Тест 3: Цикл
        const test3 = {
            code: "for i in [1, 2, 3] {\n    print('Number:', i)\n}"
        };
        console.log('\n📤 Отправка теста 3:', JSON.stringify(test3));
        ws.send(JSON.stringify(test3));
    } else if (testCount === 3) {
        // Тест 4: Функция
        const test4 = {
            code: "fn greet(name) {\n    return 'Hello, ' + name + '!'\n}\nprint(greet('DataCode'))"
        };
        console.log('\n📤 Отправка теста 4:', JSON.stringify(test4));
        ws.send(JSON.stringify(test4));
    } else {
        ws.close();
    }
});

ws.on('error', function error(err) {
    console.error('❌ Ошибка:', err.message);
});

ws.on('close', function close() {
    console.log('\n🔌 Соединение закрыто');
    console.log('✅ Все тесты завершены');
});

File: examples/ru/07-websocket/python/test_websocket.py
#!/usr/bin/env python3
"""
Тестовый скрипт для WebSocket сервера DataCode
Требуется: pip install websockets
"""

import asyncio
import websockets
import json

async def test_websocket():
    uri = "ws://127.0.0.1:8899"
    
    try:
        async with websockets.connect(uri) as websocket:
            print("✅ Подключено к серверу")
            
            # Тест 1: Простой вывод
            test1 = {
                "type": "execute",
                "code": "print('Hello, World!')"
            }
            print(f"\n📤 Отправка теста 1: {json.dumps(test1)}")
            await websocket.send(json.dumps(test1))
            
            response = await websocket.recv()
            result = json.loads(response)
            print(f"\n📥 Получен ответ:")
            print(f"  Success: {result['success']}")
            print(f"  Output: {result['output']}")
            if result.get('error'):
                print(f"  Error: {result['error']}")
            
            # Тест 2: Переменные
            test2 = {
                "type": "execute",
                "code": "global x = 10\nglobal y = 20\nprint('Sum:', x + y)"
            }
            print(f"\n📤 Отправка теста 2: {json.dumps(test2)}")
            await websocket.send(json.dumps(test2))
            
            response = await websocket.recv()
            result = json.loads(response)
            print(f"\n📥 Получен ответ:")
            print(f"  Success: {result['success']}")
            print(f"  Output: {result['output']}")
            
            # Тест 3: Цикл
            test3 = {
                "type": "execute",
                "code": "for i in [1, 2, 3] {\n    print('Number:', i)\n}"
            }
            print(f"\n📤 Отправка теста 3: {json.dumps(test3)}")
            await websocket.send(json.dumps(test3))
            
            response = await websocket.recv()
            result = json.loads(response)
            print(f"\n📥 Получен ответ:")
            print(f"  Success: {result['success']}")
            print(f"  Output: {result['output']}")
            
            # Тест 4: Функция
            test4 = {
                "type": "execute",
                "code": "fn greet(name) {\n    return 'Hello, ' + name + '!'\n}\nprint(greet('DataCode'))"
            }
            print(f"\n📤 Отправка теста 4: {json.dumps(test4)}")
            await websocket.send(json.dumps(test4))
            
            response = await websocket.recv()
            result = json.loads(response)
            print(f"\n📥 Получен ответ:")
            print(f"  Success: {result['success']}")
            print(f"  Output: {result['output']}")
            
            # Тест 5: Ошибка (для проверки обработки ошибок)
            test5 = {
                "type": "execute",
                "code": "print(undefined_variable)"
            }
            print(f"\n📤 Отправка теста 5 (ожидаем ошибку): {json.dumps(test5)}")
            await websocket.send(json.dumps(test5))
            
            response = await websocket.recv()
            result = json.loads(response)
            print(f"\n📥 Получен ответ:")
            print(f"  Success: {result['success']}")
            print(f"  Output: {result['output']}")
            if result.get('error'):
                print(f"  Error: {result['error']}")
            
            print("\n✅ Все тесты завершены")
            
    except websockets.exceptions.ConnectionRefused:
        print("❌ Ошибка: Не удалось подключиться к серверу")
        print("💡 Убедитесь, что сервер запущен: datacode --websocket --host 0.0.0.0 --port 8899")
    except Exception as e:
        print(f"❌ Ошибка: {e}")

if __name__ == "__main__":
    asyncio.run(test_websocket())

